### PR TITLE
[Debugger] TVM graph runtime memory footprint breakdown.

### PR DESCRIPTION
**Pull request by anijain2305**
As title.

For Resnet50, the output looks like this

~~~
[23:33:59] /home/ubuntu/workplace/tvm/t1/tvm/src/runtime/graph/debug/graph_runtime_debug.cc:189: Total memory allocation = 112698432 Bytes
[23:33:59] /home/ubuntu/workplace/tvm/t1/tvm/src/runtime/graph/debug/graph_runtime_debug.cc:190: Primary inputs (params + data) memory allocation = 102859936 (91%) Bytes
[23:33:59] /home/ubuntu/workplace/tvm/t1/tvm/src/runtime/graph/debug/graph_runtime_debug.cc:192: Intermediate feature maps memory allocation = 9838496 (9%) Bytes
~~~

@wweic @yzhliu @tqchen @shoubhik

### CODE REVIEW

File: src/runtime/graph/debug/graph_runtime_debug.cc [hunk 1]
```diff
@@ -136,29 +136,63 @@ class GraphRuntimeDebug : public GraphRuntime {
...
-  CHECK_LT(static_cast<size_t>(index), op_execs_.size());
-  uint32_t eid = index;
+  /*!
+   * \brief Copy index-th node to data_out.
+   *
+   * This method will do a partial run of the the graph
+   * from begining upto the index-th node and return output of index-th node.
+   * This is costly operation and suggest to use only for debug porpose.
+   *
+   * \param index: The  index of the node.
```
Comment (wweic) [inline src/runtime/graph/debug/graph_runtime_debug.cc]:
```suggestion
   * \param index: The index of the node.
```

File: src/runtime/graph/debug/graph_runtime_debug.cc [hunk 2]
```diff
@@ -136,29 +136,63 @@ class GraphRuntimeDebug : public GraphRuntime {
...
+      auto eid = entry_id(nid, 0);
+      auto storage_id = attrs_.storage_id[eid];
+      primary_inputs_size += GetDataSize(*(storage_pool_[storage_id].operator->()));
+    }
+    float percent_primary_inputs =
+        std::round(static_cast<float>(primary_inputs_size) / total_size * 100);
+
+    // Calculate intermediate feature map size.
+    uint32_t inter_size = total_size - primary_inputs_size;
+    float percent_inter = std::round(static_cast<float>(inter_size) / total_size * 100);
```
Comment (wweic) [inline src/runtime/graph/debug/graph_runtime_debug.cc]:
```suggestion
    float percent_inter = 1.0 - percent_primary_inputs;
```

**Comment by tqchen** [comment]
Let us spend a bit time thinking about the format of debugger output. Something more than logger is preferred here. We want to have a structured version(perhaps in json) that can be used by other visualizer, and a vis format that shows the result.

**Comment by soiferj** [comment]
+1 to @tqchen's comment. We can already visualize the graph JSON in something like [Netron](https://github.com/lutzroeder/Netron). It would be great to add this debug info to a debug graph JSON that can be visualized similarly. We can include runtimes and memory usage here.

**Comment by tqchen** [comment]
Close due to inactive status, I think this is a thread worth following up. with an RFC for debugger output schema and interaction with existing visualizers

**Pull request closed without merging**